run_kraken.py

```diff
@@ -37,9 +39,39 @@ def main(cfg):
     logging.info("Starting kraken")
 
     # Parse and read the config
-    if os.path.isfile(cfg):
-        with open(cfg, "r") as f:
-            config = yaml.full_load(f)
+    # Initializing flag variable as 0 to test config is a file or not
+    flag = 0
+
+    try:
+        # Checking config is yaml file or not
+        check_yaml_file = urlparse(cfg)
+        path = check_yaml_file.path
+
+        if path[-5:] == ".yaml" or path[-4:] == ".yml":
+
+            # For config as local file
+            if os.path.isfile(cfg):
+                # Set flag as 1 since config is local file
+                flag = 1
+                with open(cfg, "r") as f:
+                    config = yaml.full_load(f)
+
+            # For config as remote file
+            else:
+                # Set flag as 1 since config is remote file
+                flag = 1
```

Comment:
if you want to use flag, it should be set after the yaml is loaded.

Reply:
I agree, but when it is already checked that the config file is of yaml type, then either the file is local or remote. Once flag is set as 1, after that yaml will be loaded.